bin/NativeTests/JsRTApiTest.cpp

```diff
@@ -521,15 +521,38 @@ namespace JsRTApiTest
         REQUIRE(dbl == 3.141592);
 
         int intValue;
+        int64_t int64Value;
+
         REQUIRE(JsNumberToInt(value, &intValue) == JsNoError);
         CHECK(3 == intValue);
+        REQUIRE(JsNumberToInt64(value, &int64Value) == JsNoError);
+        CHECK(3 == int64Value);
 
         REQUIRE(JsDoubleToNumber(2147483648.1, &value) == JsNoError);
         REQUIRE(JsNumberToInt(value, &intValue) == JsNoError);
         CHECK(INT_MIN == intValue);
+        REQUIRE(JsNumberToInt64(value, &int64Value) == JsNoError);
+        CHECK(2147483648 == int64Value);
+
         REQUIRE(JsDoubleToNumber(-2147483649.1, &value) == JsNoError);
         REQUIRE(JsNumberToInt(value, &intValue) == JsNoError);
         CHECK(2147483647 == intValue);
+        REQUIRE(JsNumberToInt64(value, &int64Value) == JsNoError);
+        CHECK(-2147483649ll == int64Value);
+
+        // Number.MAX_SAFE_INTEGER
+        REQUIRE(JsDoubleToNumber(9007199254740991, &value) == JsNoError);
+        REQUIRE(JsNumberToInt(value, &intValue) == JsNoError);
+        CHECK(-1 == intValue);
+        REQUIRE(JsNumberToInt64(value, &int64Value) == JsNoError);
+        CHECK(9007199254740991 == int64Value);
+
+        // Number.MIN_SAFE_INTEGER
+        REQUIRE(JsDoubleToNumber(-9007199254740991, &value) == JsNoError);
+        REQUIRE(JsNumberToInt(value, &intValue) == JsNoError);
+        CHECK(1 == intValue);
```

Comment:
Mild preference for defining Pos_InvalidInt64/Neg_InvalidInt64 as JSRT constants, and compare against that?

Reply:
In this case it's not actually invalid, it's the result of overflowing the int32, I'm still trying to figure out what to do for truly invalid numbers that exceed the bounds of int64, those end up returning `0x8000000000000000` on x86/x64, but that doesn't seem to be 100% consistent across architectures.